File: Copilot/2.Estructura de Datos Pila (Stack)/c++/pila.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
    stack<string> pila;
    string opcion, valor;

    while (true) {
        cout << "Elige una opción (push, pop, salir): ";
        cin >> opcion;
        if (opcion == "push") {
            cout << "Valor a agregar: ";
            cin >> valor;
            pila.push(valor);
        } else if (opcion == "pop") {
            if (!pila.empty()) {
                cout << "Elemento eliminado: " << pila.top() << endl;
                pila.pop();
            } else {
                cout << "Pila vacía" << endl;
            }
        } else if (opcion == "salir") {
            break;
        }
    }
    return 0;
}

File: Copilot/2.Estructura de Datos Pila (Stack)/ruby/pila.rb
class Stack
  def initialize
    @items = []
  end

  def push(item)
    @items.push(item)
  end

  def pop
    @items.empty? ? "Pila vacía" : @items.pop
  end

  def empty?
    @items.empty?
  end
end

stack = Stack.new
loop do
  print "Elige una opción (push, pop, salir): "
  op = gets.chomp
  case op
  when "push"
    print "Valor a agregar: "
    stack.push(gets.chomp)
  when "pop"
    puts "Elemento eliminado: #{stack.pop}"
  when "salir"
    break
  end
end
